mars/services/task/execution/ray/executor.py

```diff
@@ -78,29 +81,72 @@ def execute_subtask(
     subtask_id: str,
     subtask_chunk_graph: ChunkGraph,
     output_meta_keys: Set[str],
-    input_keys: List[str],
     *inputs,
 ):
-    logger.info("Begin to execute subtask: %s", subtask_id)
+    """The function used for execute subtask in ray task.
+    If subtask is shuffle mapper, no chunk meta will be returner, otherwise return chunk mata."""
     ensure_coverage()
     subtask_chunk_graph = deserialize(*subtask_chunk_graph)
-    # inputs = [i[1] for i in inputs]
-    context = RayExecutionWorkerContext(
-        RayTaskState.gen_name(task_id), zip(input_keys, inputs)
+    subtask_digraph = subtask_chunk_graph.to_dot()
+    logger.info(
+        "Begin to execute subtask %s", subtask_id, subtask_digraph
     )
     # optimize chunk graph.
     subtask_chunk_graph = _optimize_subtask_graph(subtask_chunk_graph)
-    # from data_key to results
+    start_chunks = list(subtask_chunk_graph.iter_indep())
+    maybe_mapper_chunk = subtask_chunk_graph.result_chunks[0]
+    is_mapper = (
+        isinstance(maybe_mapper_chunk.op, MapReduceOperand)
+        and maybe_mapper_chunk.op.stage == OperandStage.map
+    )
+    if isinstance(start_chunks[0].op, PushShuffle):
+        assert len(start_chunks) == 1, start_chunks
+        # the subtask is a reducer subtask
+        num_mappers = len(inputs)
+        # single reducer may have multiple output chunks, see `PSRSshuffle._execute_reduce` and
+        # https://user-images.githubusercontent.com/12445254/168569524-f09e42a7-653a-4102-bdf0-cc1631b3168d.png
+        reducer_chunks = subtask_chunk_graph.successors(start_chunks[0])
+        reducer_operands = set(c.op for c in reducer_chunks)
+        assert len(reducer_operands) == 1, (
+            reducer_operands,
+            reducer_chunks,
+            subtask_digraph,
+        )
+        reducer_operand = reducer_chunks[0].op
+        reducer_index = reducer_operand.reducer_index
+        # mock input keys, keep this in sync with `MapReducerOperand#_iter_mapper_key_idx_pairs`
+        input_keys = [(i, reducer_index) for i in range(num_mappers)]
```

Comment:
Any tests for the input keys that matched all the cases of shuffle operands?

Reply:
Yes, will add test cases to cover shuffle operands. we definitely need do this, otherwise the shuffle implementation will be easily broke

Reply:
There is no easy way for a unit test cases cover all shuffle operands. I added assert in `execute_subtask`. The runtime  cost is minimal, I think it's acceptable.

Reply:
Can you add some assertion to the `CheckedSubtaskProcessor` to make sure the map keys you generated is exactly the same with Mars original logic.

Reply:
The keys will generated only when `PushShuffle` is used. Mars dones't build `PushShuffle` into subtask graph.